projects/igniteui-angular/src/lib/grids/grid/grid.component.ts

```diff
@@ -189,6 +189,7 @@ export class IgxGridComponent extends IgxGridBaseComponent implements OnInit, Do
             ungroupedColsArr.forEach((elem) => {
                 ungroupedCols.push(this.getColumnByName(elem.fieldName));
             }, this);
+            this.ngDoCheck();
```

Comment:
@PavlovVasil Instead of directly calling the lifecycle ngDoCheck handler It would be better to call detectChanges instead.